src/material-experimental/mdc-form-field/directives/notched-outline.ts

```diff
@@ -34,30 +35,42 @@ import {MDCNotchedOutline} from '@material/notched-outline';
   encapsulation: ViewEncapsulation.None,
 })
 export class MatFormFieldNotchedOutline implements AfterViewInit, OnDestroy {
-  private _mdcNotchedOutline: MDCNotchedOutline;
+  private _mdcNotchedOutline: MDCNotchedOutline|null = null;
 
-  constructor(private _elementRef: ElementRef) {}
+  constructor(private _elementRef: ElementRef, private _platform: Platform) {}
 
   ngAfterViewInit() {
-    // The notch component relies on the view to be initialized. This means
-    // that we cannot extend from the "MDCNotchedOutline".
-    this._mdcNotchedOutline = MDCNotchedOutline.attachTo(this._elementRef.nativeElement);
+    // The notched outline cannot be attached in the server platform. It schedules tasks
+    // for the next browser animation frame and relies on element client rectangles to render
+    // the outline notch. To avoid failures on the server, we just do not initialize it,
+    // but the actual notched-outline styles will be still displayed.
+    if (this._platform.isBrowser) {
+      // The notch component relies on the view to be initialized. This means
+      // that we cannot extend from the "MDCNotchedOutline".
+      this._mdcNotchedOutline = MDCNotchedOutline.attachTo(this._elementRef.nativeElement);
+    }
   }
 
   ngOnDestroy() {
-    this._mdcNotchedOutline.destroy();
+    if (this._mdcNotchedOutline !== null) {
```

Comment:
Can you just check `if (this._mdcNotchedOutline)`? That would prevent needing the `!`

(here and elsewhere)

Reply:
Ops. I actually didn't want to keep the `!`. Before I just used a `this._assertXX` call, and needed the `!`. But I went back to an explicit if clause.